internal/ci/ci.go

```diff
@@ -101,138 +100,127 @@ func GetCI(provider Provider) CI {
 	switch provider {
 	case AppVeyor:
 		return CI{
-			Provider:  provider,
-			OriginURL: fmt.Sprintf("%s/project/%s/%s/builds/%s", os.Getenv("APPVEYOR_URL"), os.Getenv("APPVEYOR_ACCOUNT_NAME"), os.Getenv("APPVEYOR_PROJECT_NAME"), os.Getenv("APPVEYOR_BUILD_ID")),
-			Repo:      os.Getenv("APPVEYOR_REPO_NAME"),
-			RefName:   os.Getenv("APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH"),
-			SHA:       os.Getenv("APPVEYOR_REPO_COMMIT"),
+			Provider: provider,
+			URL:      fmt.Sprintf("%s/project/%s/%s/builds/%s", os.Getenv("APPVEYOR_URL"), os.Getenv("APPVEYOR_ACCOUNT_NAME"), os.Getenv("APPVEYOR_PROJECT_NAME"), os.Getenv("APPVEYOR_BUILD_ID")),
+			Repo:     os.Getenv("APPVEYOR_REPO_NAME"),
+			Ref:      os.Getenv("APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH"),
+			ShortSHA: shortenSHA(os.Getenv("APPVEYOR_REPO_COMMIT")),
 		}
 	case AWS:
 		return CI{
-			Provider:  provider,
-			OriginURL: os.Getenv("CODEBUILD_PUBLIC_BUILD_URL"),
-			Repo:      os.Getenv("CODEBUILD_SOURCE_REPO_URL"),
-			RefName:   os.Getenv("CODEBUILD_SOURCE_VERSION"),
-			SHA:       os.Getenv("CODEBUILD_RESOLVED_SOURCE_VERSION"),
-			User:      os.Getenv("CODEBUILD_WEBHOOK_ACTOR_ACCOUNT_ID"),
+			Provider: provider,
+			URL:      os.Getenv("CODEBUILD_PUBLIC_BUILD_URL"),
+			Repo:     os.Getenv("CODEBUILD_SOURCE_REPO_URL"),
+			Ref:      os.Getenv("CODEBUILD_SOURCE_VERSION"),
+			ShortSHA: shortenSHA(os.Getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")),
 		}
 	case Azure:
 		return CI{
-			Provider:  provider,
-			OriginURL: os.Getenv("Build_BuildUri"),
-			Repo:      os.Getenv("System_PullRequest_SourceRepositoryURI"),
-			RefName:   os.Getenv("Build_SourceBranchName"),
-			SHA:       os.Getenv("Build_SourceVersion"),
-			User:      os.Getenv("Build_RequestedFor"),
+			Provider: provider,
+			URL:      os.Getenv("BUILD_REPOSITORY_URI"),
+			Repo:     os.Getenv("SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI"),
+			Ref:      os.Getenv("BUILD_SOURCEBRANCHNAME"),
+			ShortSHA: shortenSHA(os.Getenv("BUILD_SOURCEVERSION")),
 		}
 	case Bamboo:
 		return CI{
-			Provider:  provider,
-			OriginURL: os.Getenv("bamboo_buildResultsUrl"),
-			Repo:      os.Getenv("bamboo_planRepository_repositoryUrl"),
```

Comment:
Terminologies: In Bamboo CI, a `plan` organizes a series of jobs and their associated configurations. `position` is the index of a repository in the plan's configuration. The default position is `0`.

The variable `bamboo_planRepository_repositoryUrl` is now replaced with `bamboo_planRepository_<position>_repositoryUrl`. There’s no way to dynamically retrieve `position`.
For commit SHA information, the only available variable is `bamboo_planRepository_<position>_revision`, which also depends on `<position>`. The value cannot be accurately determined.

Based on these, I removed the repo url and revision.

Reference: https://confluence.atlassian.com/bamboo/bamboo-variables-289277087.html

Note: How can these variables be accessed as environment variables?

> Bamboo variables are exported as bash shell variables. All full stops (periods) are converted to underscores. For example, the variable bamboo.my.variable is $bamboo_my_variable in bash. This is related to File Script and Inline Script tasks.